src/Stratis.Bitcoin/P2P/SelfEndpointTracker.cs

```diff
@@ -22,5 +39,41 @@ public bool IsSelf(IPEndPoint ipEndPoint)
         {
             return this.knownSelfEndpoints.Contains(ipEndPoint);
         }
+
+        /// <summary>Updates external IP address and peer score of the node.</summary>
+        public void UpdateAndAssignMyExternalAddress(IPEndPoint ipEndPoint, int? ipEndPointScore = null)
+        {
```

Comment:
And we just introduce logs into this method

Reply:
Done but I didn't alter all the calling code with logmanagers as wanted to confirm before affecting a lot of files.